libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -50,12 +52,23 @@ public class MediaLine : ISerializationCallbackReceiver
         /// If <see cref="Transceiver"/> is valid, that is a first session negotiation has already been completed,
         /// then changing this value raises a <see xref="WebRTC.PeerConnection.RenegotiationNeeded"/> event on the
         /// peer connection of <see cref="Transceiver"/>.
-        /// </summary>
+        ///
+        /// Must be changed on the main thread.
+        /// </remarks>
         public MediaTrackSource Source
         {
             get { return (_source as MediaTrackSource); }
             set
             {
+                // If the connection is active, ensure that this is run on the main thread
+                // and doesn't race with the signaling handlers.
+                // If the connection is not awake we cannot use EnsureIsMainAppThread, but
+                // there should be no race on shared state, so skip the check.
+                if (_peer.isActiveAndEnabled)
+                {
+                    _peer.EnsureIsMainAppThread();
+                }
```

Comment:
This makes little sense, you can only access `isActiveAndEnabled` from the main Unity app thread, otherwise you get an exception I believe.

Reply:
Conflating into https://github.com/microsoft/MixedReality-WebRTC/pull/458#discussion_r454987005